### PR TITLE
pidgin: now builds and runs on osx

### PR DESCRIPTION
**Pull request by fkautz**
Signed-off-by: Frederick F. Kautz IV <fkautz@alumni.cmu.edu>

A quick note: this is my first patch in nixpkgs. I don't know if I have followed best practices. E.g. I read sandboxing is enabled by default but I don't know how to confirm it.

Also, this package does require an X11 server to run. Nix does provide access to the xquartz installer, but it is possible to run pidgin with an already running X11 server.

###### Motivation for this change
Before: No Pidgin on OSX
After: Pidgin on OSX

###### Things done

Changed build to support unix.

* Build: constrain to platforms.unix instead of platforms.linux
* Build: Differences Linux and OSX now parameterized
* Darwin: build against gtk2-x11 instead of gtk
* Darwin: disable gtkspell2 (spellcheck)
* Darwin: disable farstream (voice and video)

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
  - I didn't disable sandbox, I read somewhere in docs that it is enabled by default. Don't know how to confirm.
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - I see that these tests exist, I don't know how to run them yet. Would appreciate assistance here.
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
  - [x] OSX
  - [ ] Linux (I have a headless NixOS)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
  - No docs found
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
  - I read through and attempted to adapt. Please verify.

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/pidgin/default.nix [hunk 1]
```diff
@@ -54,6 +62,8 @@ let unwrapped = stdenv.mkDerivation rec {
     "--disable-meanwhile"
     "--disable-nm"
     "--disable-tcl"
+    "--disable-gtkspell"
+    "--disable-vv"
```
Comment (vcunat) [inline pkgs/applications/networking/instant-messengers/pidgin/default.nix:66]: The combination of always disabling gtkspell with sometimes passing it is weird, but it's possible that overall the flags need revision.

Comment (fkautz) [reply pkgs/applications/networking/instant-messengers/pidgin/default.nix:66]: Good catch, these need to be parameterized based on environment. I'll update this patch.

**Comment by Mic92** [comment]
Not quite sure if I need to disable the sandbox?

```
[nix-shell:~/.cache/nixpkgs-review/pr-91882-1]$ nix log /nix/store/1nywh7sccp8l0x4f7wk2n5nkyi855v71-libstartup-notification-0.12.drv
warning: unknown setting 'experimental-features'
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ STDIN
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ @nix { "action": "setPhase", "phase": "unpackPhase" }
   2   │ unpacking sources
   3   │ unpacking source archive /nix/store/bbqnd6qxw97v4dh2dxlysp0kckfvdj51-startup-notification-0.12.tar.gz
   4   │ dyld: Library not loaded: @rpath/CoreFoundation.framework/Versions/A/CoreFoundation
   5   │   Referenced from: /nix/store/bzf81p8lczllrs1633zaj0rka7cz4k8g-gettext-0.20.1/lib/libintl.8.dylib
   6   │   Reason: no suitable image found.  Did find:
   7   │     /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation: file system sandbox blocked stat()
   8   │ /nix/store/swr2h40348gmxmsxwfyhlm6xsrjyw0cj-stdenv-darwin/setup: line 838:  1661 Abort trap: 6           tar xf "$fn"
   9   │ do not know how to unpack source archive /nix/store/bbqnd6qxw97v4dh2dxlysp0kckfvdj51-startup-notification-0.12.tar.gz
```

**Comment by fkautz** [comment]
@Mic92 was that posted to the correct PR? This code shouldn't affect anything in that library.

**Comment by fkautz** [comment]
@vcunat I've ran `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` and found a problem with a downstream purple-matrix library. libpurple is pidgin's library for building IM messages on. It's in pidgin-plugins. I have a fix for it that I intend to submit after this one since it's a different package.

**Comment by Mic92** [comment]
> @Mic92 was that posted to the correct PR? This code shouldn't affect anything in that library.

Well startup-notifications is one of pidgin's dependencies.

**Comment by fkautz** [comment]
@Mic92 what command did you use to reproduce this? I'm very new here so still trying to navigate the tooling. :)

I was just using `nix-build -A pidgin` and `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`

And would this be the appropriate PR to fix that problem or should I open a new PR?

**Comment by Mic92** [comment]
> @Mic92 what command did you use to reproduce this? I'm very new here so still trying to navigate the tooling. :)
> 
> I was just using `nix-build -A pidgin` and `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
> 
> And would this be the appropriate PR to fix that problem or should I open a new PR?

You can reproduce it by `sandbox = true` in `$HOME/.config/nix/nix.conf` or in your system-config nix.conf if you are an unprivileged user on a multi-user nix installation (in other words when you have a nix-daemon running). I don't know what are our policies are for macOS right now and if hydra has sandboxes enabled or not. I am only using a macOS vm occasionally and just copied my configuration over from Linux. For now I just trust of ofborg to be the ground truth: @GrahamcOfBorg build pidgin

**Pull request merged**